### PR TITLE
[fix] Dynamic updating of notification verb #334

### LINKED ISSUE #334: [bug] Notification verb is saved in database
**Describe the bug**
The notification `verb` is saved in the database. Thus, changing the `verb` in the `NOTIFICATION_TYPE` does not change the message of existing notifications.

**Steps To Reproduce**
1. Create a default notification using the [`create_notification` management command](https://openwisp.io/docs/dev/notifications/user/management-commands.html#create-notification). 
2. Go to Django admin and verify the notification message. 
```
Default notification with default verb and level info by default
```
3. Change the verb for the default notification type 
https://github.com/openwisp/openwisp-notifications/blob/55a3443956f2ffa0301bd44a72d051f280fd02f7/openwisp_notifications/types.py#L7-L19
4. Go to Django admin again and verify the notification message 

**Expected behavior**
The notification message should show the updated notification verb 

**Possible solution**

Update the notify handler to not save the verb in the database. 

https://github.com/openwisp/openwisp-notifications/blob/55a3443956f2ffa0301bd44a72d051f280fd02f7/openwisp_notifications/handlers.py#L139-L148

Update the AbstractNotification model to use the `verb` from the configuration if database value is set to None

### PR DESCRIPTION
**Pull request by pranshustuff**
## Checklist

- [ ] I have read the [OpenWISP Contributing Guidelines](http://openwisp.io/docs/developer/contributing.html).
- [ ] I have manually tested the changes proposed in this pull request.
- [ ] I have written new test cases for new code and/or updated existing tests for changes to existing code.
- [ ] I have updated the documentation.

## Reference to Existing Issue

Fixes #334.

## Description of Changes

Made verb field in notification nullable. Created
a property verb under AbstractNotification in models that prioritizes verb from NOTIFICATION_TYPES and
uses DB as a fallback.

## Screenshot

https://github.com/user-attachments/assets/c8a8b1de-778e-43cc-b541-5314ed9f8d31

### CODE REVIEW

File: openwisp_notifications/base/models.py [hunk 1]
```diff
@@ -101,6 +101,18 @@ class AbstractNotification(UUIDModel, BaseNotification):
     _action_object = BaseNotification.action_object
     _target = BaseNotification.target
 
+    @property
+    def verb(self):
+        try:
+            config = get_notification_configuration(self.type)
+            return config.get("verb") or self.__dict__.get("verb") or "unspecified"
```
Comment (nemesifier) [inline openwisp_notifications/base/models.py:108]: where is "unspecified" coming from?

**Comment by pranshustuff** [comment]
I think the CI builds are failing because the ./manage.py makemigrations isn't being run in the workflow, the migration I added solved the issue of the NOT NULL constraint on verb when I tested locally.

File: openwisp_notifications/base/models.py (hunk 1 above)
Comment (nemesifier) [reply openwisp_notifications/base/models.py:108]: @pranshustuff see https://www.linfo.org/rule_of_silence.html

File: openwisp_notifications/base/models.py [hunk 2]
```diff
@@ -101,6 +101,18 @@ class AbstractNotification(UUIDModel, BaseNotification):
...
+        except Exception:
```
[the remark this answers is not in the document]
Comment (pranshustuff) [reply openwisp_notifications/base/models.py:109]: Probably a NotificationRenderException as mentioned in handlers.py is removed or a TypeError if self.type is something like a list, just to cover edge cases.

**Review by nemesifier — CHANGES_REQUESTED** [review]
Tests and QA checks are failing, the [developer docs of OpenWISP RADIUS explain how to run tests and QA checks](https://openwisp.io/docs/dev/radius/developer/installation.html).

**Comment by pranshustuff** [comment]
Ok locally, it passed the tests, on sample app too.

File: openwisp_notifications/base/models.py [hunk 3]
```diff
@@ -101,6 +101,21 @@ class AbstractNotification(UUIDModel, BaseNotification):
...
+    @property
+    def verb(self):
+        config = {}
+        try:
+            config = get_notification_configuration(self.type)
+        except (NotificationRenderException, TypeError) as e:
+            logger.error(
+                "Couldn't get notification config for type %s : %s", self.type, e
+            )
+        return config.get("verb") or self.__dict__.get("verb")
```
[the remark this answers is not in the document]
Comment (pranshustuff) [reply openwisp_notifications/base/models.py]: using self.verb now.

File: openwisp_notifications/base/models.py [hunk 4]
```diff
@@ -101,6 +101,21 @@ class AbstractNotification(UUIDModel, BaseNotification):
...
+
+    @verb.setter
+    def verb(self, value):
+        self.__dict__["verb"] = value
```
[the remark this answers is not in the document]
Comment (pranshustuff) [reply openwisp_notifications/base/models.py]: Setter was a leftover from a previous attempt. Not needed anymore.
